cpp/src/plasma/client.cc

```diff
@@ -604,10 +604,16 @@ Status PlasmaClient::Seal(const ObjectID& object_id) {
   // Make sure this client has a reference to the object before sending the
   // request to Plasma.
   auto object_entry = objects_in_use_.find(object_id);
-  ARROW_CHECK(object_entry != objects_in_use_.end())
-      << "Plasma client called seal an object without a reference to it";
-  ARROW_CHECK(!object_entry->second->is_sealed)
-      << "Plasma client called seal an already sealed object";
+
+  if (object_entry == objects_in_use_.end()) {
+    return Status::PlasmaObjectNonexistent(
+        "Plasma client called seal an object without a reference to it");
```

Comment:
I'm not a native English speaker, but while we're at it, perhaps we could fix the grammar in this error message and the one below?
Something like "Seal() called on an object without a reference to it".

Reply:
Me neither :) Shouldn't we incorporate the object ID in the message?

Reply:
If there's a function to pretty-print object IDs when why not :-)

Reply:
@pitrou I guess pretty printing of plasma object should follow the API of [arrow::pretty_print](https://github.com/apache/arrow/blob/master/cpp/src/arrow/pretty_print.h), if so should I create a JIRA ticket instead?

Reply:
I'm not sure, but that sounds reasonable, and can be discussed on a JIRA issue indeed.